### PR TITLE
CMP app: fix `-reqin` option

### PR DESCRIPTION
**Pull request by DDvO**
It turned out that the  `-reqin` option of the CMP CLI had two bugs, fixed here:
* the last `-reqin` file argument was not actually used, which covered a further bug:
* the `recipNonce` in the server response was not used in the next request.

Fixing the latter requires a new helper function at API level: `OSSL_CMP_MSG_update_recipNonce`.

On this occasion further corrected and slightly extended the related CLI and API docs.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: util/libcrypto.num [hunk 1]
```diff
@@ -5449,6 +5449,7 @@ OSSL_CMP_CTX_get0_libctx                ?	3_2_0	EXIST::FUNCTION:CMP
 OSSL_CMP_CTX_get0_propq                 ?	3_2_0	EXIST::FUNCTION:CMP
 OSSL_CMP_CTX_reset_geninfo_ITAVs        ?	3_0_8	EXIST::FUNCTION:CMP
 OSSL_CMP_CTX_get0_validatedSrvCert      ?	3_2_0	EXIST::FUNCTION:CMP
+OSSL_CMP_MSG_update_recipNonce          ?	3_0_8	EXIST::FUNCTION:CMP
```
Comment (t8m) [inline util/libcrypto.num:5452]: This won't make it into 3.0.8.

**Review by t8m — COMMENTED** [review]
I am not sure if this is a true bug fix. Is the -reqin feature fully broken or just does not support some use cases?

File: util/libcrypto.num (hunk 1 above)
Comment (DDvO) [reply util/libcrypto.num]: Changed to 3.0.9.

**Comment by DDvO** [comment]
> I am not sure if this is a true bug fix. Is the -reqin feature fully broken or just does not support some use cases?

It is broken in all uses:
If just one file with request is given, this file is ignored.
If several files are given, the last one is ignored.

And when the above is fixed, the other bug would cause a server error
for all transactions that contain more than one request and response.

So the feature is fully broken, unfortunately.

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
ping @openssl/committers for 2nd review

**Comment by DDvO** [comment]
Merged to master, 3.0, and 3.1 - thanks @t8m and @paulidale

**Pull request closed without merging**